src/components/DatasetCreateView.jsx

```diff
@@ -139,7 +139,9 @@ export class DatasetCreateView extends React.PureComponent {
 
   addUser(newEmail) {
     const { dispatch, readers } = this.props;
-    if (!_.includes(readers, newEmail)) {
+    if (!readers) {
+      dispatch(actions.change('dataset.readers', [newEmail]));
+    } else if (!_.includes(readers, newEmail)) {
```

Comment:
Does this dispatch in the case where readers includes newEmail?

Reply:
Where it does not